tests/components/Microsoft.DotNet.UpgradeAssistant.MSBuild.Tests/MSBuildTestBuild.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.DotNet.UpgradeAssistant.MSBuild.Tests
+{
+    public sealed class MSBuildTestBuild : IDisposable
+    {
+        public string TestFolderPath { get; }
+
+        public MSBuildTestBuild(string path)
+        {
+            TestFolderPath = path;
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+        }
+
+        public string Add(string fileName, string fileContents)
+        {
+            string file = Path.Combine(TestFolderPath, fileName);
+            using (var sw = new StreamWriter(file))
+            {
+                sw.WriteLine(fileContents);
+            }
+
+            return file;
+        }
+
+        public static Microsoft.Build.Evaluation.Project Build(string path)
+        {
+            using var collection = new ProjectCollection();
```

Comment:
not sure if you want to do a `using` here as it will dispose of things in the collection. Better probably to track that in the builder object and dispose of it with the directory (and ensure no double calling the Build method as a sanity check, since you'll only want a single collection to be tracked)